src/lightning_lite/strategies/launchers/subprocess_script.py

```diff
@@ -142,13 +141,16 @@ def _check_can_spawn_children(self) -> None:
 
 
 def _basic_subprocess_cmd() -> Sequence[str]:
+    import __main__  # local import to avoid https://github.com/Lightning-AI/lightning/issues/15218
```

Comment:
Is there a reason why you didn't include the repro in the issue as a test?

Maybe because it would need to be marked as standalone?

Reply:
If you run with `python -i`, it will drop you in a shell. The test would never pass because it would wait for user input.

Reply:
The repro could be included as a standalone subprocess call, but it won't trigger a failure. The code that errors in the repro runs in the atexit handler. The return code of the subprocess is 0, so you won't get a test failure when a regression occurs. This is what I found.